src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -821,9 +830,31 @@ private[internal] trait TypeMaps {
       mapTreeSymbols.transform(tree)
   }
 
+  /** A map to implement the `substSym` method. */
+  class SubstSymMap(override val from: List[Symbol], override val to: List[Symbol]) extends AbstractSubstSymMap {
+    reload()
+
+    def this(pairs: (Symbol, Symbol)*) = this(pairs.toList.map(_._1), pairs.toList.map(_._2))
```

Comment:
If walking the from/to in the assert is expensive, is pairs.toList.map twice also expensive? No, I see it's just called from Refchecks with one pair.

Reply:
Note that this constructor, for the immutable class, was already defined (above). It is only called from one location.